gps_ubxm8_i2c/ubxPacket.cpp

```diff
@@ -0,0 +1,69 @@
+#include "ubxPacket.h"
+
+// make a ubxPacket class that has a method to update one byte at a time
+UBXPacketReader::UBXPacketReader() {
+    payloadLength = 0;
+    packetIndex = 0;
+    ckA = 0;
+    ckB = 0;
+    complete = false;
+    inProgress = false;
+}
+
+UBXPacketUpdateResult UBXPacketReader::update(uint8_t newByte) {
+    if (complete) {
+        return UBXPacketUpdateResult::PACKET_ALREADY_COMPLETE;
+    }
+    inProgress = true;
+    int currentPacketIndex = packetIndex;
+    packetIndex++;
+    if (currentPacketIndex == 0) {
+        messageClass = newByte;
+    } else if (currentPacketIndex == 1) {
+        messageId = newByte;
+    } else if (currentPacketIndex == 2) {
+        payloadLength = newByte;
+    } else if (currentPacketIndex == 3) {
+        payloadLength |= (uint16_t)newByte << 8;
+    } else if (currentPacketIndex >= 4 + payloadLength && currentPacketIndex < 6 + payloadLength) {
```

Comment:
the order here is a bit confusing, also recommend add extra brackets for clarity, perhaps do
```c
} else if ((currentPacketIndex >= 4) && (currentPacketIndex < 4 + payloadLength)) {
// blah blah
} else if (currentPacketIndex == 4 + payloadLength) {
// blah blah
} else {
//blah blah
}
```